bundles/org.openhab.binding.radiothermostat/src/main/java/org/openhab/binding/radiothermostat/internal/discovery/RadioThermostatDiscoveryService.java

```diff
@@ -0,0 +1,301 @@
+                            logger.debug("Match: {} ", response);
+                            parseResponse(response);
+                        }
+                    }
+                    logger.debug("Bridge device scan interrupted");
+                } catch (SocketTimeoutException e) {
+                    logger.debug(
+                            "Timed out waiting for multicast response. Presumably all devices have already responded.");
+                }
+            } finally {
+                socket.leaveGroup(m);
+                socket.close();
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.debug("got exception: {}", e.getMessage());
+        }
+        return;
+    }
+
+    /**
+     * Scans all messages that arrive on the socket and scans them for the
+     * search keywords. The search is not case sensitive.
+     *
+     * @param socket
+     *            The socket where the answers arrive.
+     * @param keywords
+     *            The keywords to be searched for.
+     * @return
+     * @throws IOException
+     */
+
+    protected void parseResponse(String response) {
+        DiscoveryResult result;
+
+        String name = "unknownName";
+        String uuid = "unknownThermostat";
+        String ip = null;
+        String url = null;
+
+        Scanner scanner = new Scanner(response);
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] pair = line.split(":", 2);
+            if (pair.length != 2) {
+                continue;
+            }
+            String key = pair[0].toLowerCase();
+            String value = pair[1].trim();
+            logger.debug("key: {} value: {}.", key, value);
+            switch (key) {
+                case "location":
+                    try {
+                        url = value;
+                        ip = new URL(value).getHost();
+                    } catch (MalformedURLException e) {
+                        logger.debug("Malfored URL {}", e.getMessage());
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+        scanner.close();
+
+        logger.debug("Found thermostat, ip: {} ", ip);
+
+        if (ip == null) {
+            logger.debug("Bad Format from thermostat");
+            return;
+        }
+
+        JsonObject content;
+        String sysinfo;
+
+        try {
+            // Run the HTTP request and get the JSON response from the thermostat
+            ContentResponse contentResponse = httpClient.newRequest(url).method(GET).timeout(20, TimeUnit.SECONDS)
+                    .send();
+            sysinfo = contentResponse.getContentAsString();
+            content = new JsonParser().parse(sysinfo).getAsJsonObject();
+            uuid = content.get("uuid").getAsString();
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Comment:
Don't forget to catch `JsonSyntaxException`

Reply:
done